pkg/util/pki/generate.go

```diff
@@ -111,6 +123,18 @@ func GenerateECPrivateKey(keySize int) (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(ecCurve, rand.Reader)
 }
 
+// GenerateEdPrivateKey will generate an EdDsa private key of the given size.
+// It can be used to generate only 25519
+func GenerateEdPrivateKey(keySize int) (*ed25519.PrivateKey, error) {
+
+	if keySize != Ed25519 {
+		return nil, fmt.Errorf("unsupported EdDsa key size specified only 25519: %d", keySize)
+	}
+	_, prvkey, err := ed25519.GenerateKey(rand.Reader)
+
+	return &prvkey, err
+}
+
```

Comment:
suggestion: there's no configuration built into Ed25519 so I think the `keySize` parameter is redundant here.

We also don't need to return a pointer to a private key; ed25519.PrivateKey is a type alias for `[]byte` which implements `crypto.Signer` and can be used with a non-pointer receiver.

```suggestion
// GenerateEd25519PrivateKey generates an Ed25519 private key which can be used for TLS.
func GenerateEd25519PrivateKey() (ed25519.PrivateKey, error) {
	_, privkey, err := ed25519.GenerateKey(rand.Reader)
	return privkey, err
}

```

Reply:
I applied your feedback and also sent the test I have been building.

Would you prefer me to mark this as resolved or do you mark them as resolved?

Reply:
I don't mind who marks them as resolved really, but I might as well do it! I might not be able to get back to this today (UK time, UTC+1) but I'll try to take another look soon :+1: